R/group_data.R

```diff
@@ -63,3 +63,31 @@ add_sum_of_views <- function(df, view_name_pattern, method, new_name = view_name
   df_plus_new_group <- bind_rows(df, new_group)
   return(df_plus_new_group)
 }
+
+#' @param df data.frame of daily unique pageviews and page paths with site ids for noms
+#'           expects pagepath, date, uniquePageviews and pageviews
+#' @return data frame with date, uniquePageviews and site_no
+group_by_site_id <- function(df) {
+  
+  date <- df$date[1]
+  
+  df <- df %>%
```

Comment:
Looking into this, but I did a quick comparison and for some reason I come out with different # of rows when I combine the first  together, I don't know why though yet.

Reply:
```
df <- df %>%
    mutate(pagePath = gsub("&amp;", "&", pagePath), #clean up yucky url bits like encodings and diff param names
      pagePath = gsub("search_site_no","site_no"),
      pagePath = gsub("Site_no","site_no"),
      pagePath = gsub("multiple_site_no","site_no"),
      pagePath = gsub(", ",",")) %>%
    mutate(site_no = param_get(urls = pagePath, parameter_names = c("site_no"))) %>% #pull out site_no data in to their own column
    mutate(site_no = site_no$site_no) %>% #site_no column data character instead of df
    separate_rows(site_no, sep=",") %>% #break up multiple site nos separated by commas
    mutate(site_no = trimws(site_no, which=c("both"))) %>% #trim extraneous white space from around site_no
    filter(!(is.na(site_no) | site_no=="")) %>% #drop empty site_no rows
    separate_rows(site_no, sep=" ") %>% #if any exist, break up multiple site nos separated by spaces
    filter(grepl('^\\d+$', site_no)) %>% #only keep numeric site_no data
    filter(nchar(as.character(site_no)) >=8) %>% #keep values at 8 or more characters, likely erroneous if less
    ddply("site_no",numcolwise(sum)) %>% #add together uniquePageviews by site_no
    mutate(date = date) %>% #put date back
    select(date, site_no, uniquePageviews) #keep only columns that we need
```

Reply:
You would still need to supply `pagePath` as the third argument to `gsub` on lines 3-6 in your code block I think, unless you piped together all the `gsubs`.  I'm surprised that  doesn't error as is, its not clear what would be used for the gsub `replacement` argument.
![image](https://user-images.githubusercontent.com/13987349/118696934-a33b2480-b7c3-11eb-8975-55ca27c97e6b.png)

Reply:
Probably best to name the arguments there to be safe.

Reply:
thanks, oops. I'll check that out when I get back to this a little later.